src/lib/sync-profile.js

```diff
@@ -155,10 +163,10 @@ class SyncProfile {
 
     if (msg2 === 'alreadySynced') {
       throw new Error('alreadySynced')
-    }
-
-    if (!key || msg2 !== 'requestWriteAccess') {
-      throw new Error('msg not expected' + msg2)
+    } else if (msg2 === 'abort') {
+      throw new Error('abort')
+    } else if (!key || msg2 !== 'requestWriteAccess') {
+      throw new Error('msg not expected ' + msg2)
     }
 
     await db.authorizeAsync(Buffer.from(key, 'hex'))
```

Comment:
if key is always expected (line 161), why not using a switch statement
```suggestion
if (!key) throw new Error('no key')

    switch (msg2) {
      case 'alreadySynced':
        throw new Error('alreadySynced')
      case 'abort':
        throw new Error('abort')
      case 'requestWriteAccess':
        await db.authorizeAsync(Buffer.from(key, 'hex'))
        await sendEncryptedJSON({ msg: 'writeAccessGranted' }, this.key, this.peer)
        break
      default:
        throw new Error('msg not expected ' + msg2)
    }
```